src/main/kotlin/com/lykke/client/accounts/rabbitmq/ClientAccountsRmqListener.kt

```diff
@@ -0,0 +1,129 @@
+package com.lykke.client.accounts.rabbitmq
+
+import com.lykke.client.accounts.config.RabbitMqConfig
+import com.lykke.client.accounts.incoming.ClientAccountMessages.WalletCreatedEvent as WalletCreatedEvent
+import com.lykke.utils.logging.ThrottlingLogger
+import com.lykke.utils.rabbit.Connector
+import com.lykke.utils.rabbit.ConsumerFactory
+import com.lykke.utils.rabbit.RabbitMqSubscriber
+import com.rabbitmq.client.Channel
+import com.rabbitmq.client.ConnectionFactory
+import java.lang.Exception
+import java.lang.IllegalStateException
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.CopyOnWriteArrayList
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.function.Consumer
+import kotlin.concurrent.thread
+import com.lykke.utils.rabbit.RabbitMqConfig as UtilsRabbitMqConfig
+
+class ClientAccountsRmqListener(
+    private val rabbitMqConfig: RabbitMqConfig,
+    private val messageDeserializer: Deserializer<WalletCreatedEvent>
+) {
+
+    private companion object {
+        val LOGGER = ThrottlingLogger.getLogger(ClientAccountsRmqListener::class.java.simpleName)
+    }
+
+    private val eventHandlers = CopyOnWriteArrayList<Consumer<WalletCreatedEvent>>()
+    private val eventsQueue: BlockingQueue<ByteArray> = rabbitMqConfig.queue ?: LinkedBlockingQueue<ByteArray>()
+    private val rabbitMqSubscriber: RabbitMqSubscriber
+    private val eventProcessingThread: Thread
+    private var started = false
+
+    init {
+        rabbitMqSubscriber = getRabbitMqSubscriber()
+        eventProcessingThread = start()
+    }
+
+    fun addEventHandler(listener: Consumer<WalletCreatedEvent>) {
+        eventHandlers.add(listener)
+    }
+
+    fun close() {
+        try {
+            rabbitMqSubscriber.shutdown()
+        } catch (e: Exception) {
+            LOGGER.error("Error occurred while shutting down client accounts RMQ listener", e)
+        }
+        eventProcessingThread.interrupt()
+    }
+
+    private fun start(): Thread {
+        if (started) {
+            throw IllegalStateException("Client accounts rmq listener already stared")
+        }
+
+        rabbitMqSubscriber.start()
+        val eventProcessingThread = startEventProcessingLoop()
+
+        started = true
+
+        return eventProcessingThread
+    }
+
+    private fun startEventProcessingLoop(): Thread {
+        return thread(name = ClientAccountsRmqListener::class.java.simpleName) {
```

Comment:
Is it better to use `java.util.concurrent.Executor` passed to this class to execute the thread ?

Reply:
We can add executor, but don't see any advantage - lib will use only one the thread anyway. Also, we have a shutdown method - that stops the target thread.